connectors/src/connectors/webcrawler/temporal/activities.ts

```diff
@@ -188,16 +189,42 @@ export async function crawlWebsiteByConnectorId(connectorId: ModelId) {
     })
   );
 
-  await crawler.run([webCrawlerConfig.url]);
+  await crawler.addRequests([webCrawlerConfig.url]);
 
-  await crawler.teardown();
+  const crawlPromise = crawler.run([webCrawlerConfig.url]);
+  let isResolved = false;
+  crawlPromise
+    .then(() => {
+      isResolved = true;
+    })
+    .catch((e) => {
+      logger.error(
+        {
+          error: e,
+          connectorId: connector.id,
+          configId: webCrawlerConfig.id,
+        },
+        "Webcrawler error while crawling"
+      );
+    });
+  try {
```

Comment:
I'm not sure I understand? That means we will not process error coming from the crawler right and will only log. The activity will never fail here?

Reply:
You’ll be off the main event loop at this stage

Reply:
And the error won’t be processed as an activity error ?

Reply:
right; will update the code accordingly

Reply:
AFAIU thee doc from temporal linked above, temporal's sleep method is a way to opt in to cancel workflow signal

Reply:
Then let's add it in the activity logic ?

Reply:
anyways, I am going to test / clean this code so you can take another look then